spec/models/token_spec.rb

```diff
@@ -1,5 +1,35 @@
-require 'rails_helper'
+require "rails_helper"
 
 RSpec.describe Token, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  let(:user) { build(:user) }
+  subject { build(:token, user: user) }
+
+  context "when factory is built" do
+    it { is_expected.to be_valid }
+    it { is_expected.to be_a Token }
+  end
+
+  context "when calling the instance methods" do
+    it { is_expected.to respond_to(:user_id, :temp, :status, :set_temp_token) }
+  end
+
+  describe "Instance Methods" do
+    describe "#set_temp_token" do
+      let(:temp) do
+        subject.set_temp_token
+        subject.temp
+      end
+
+      it { expect(temp).to be_a String }
+      it { expect(temp.length).to eq(32) }
+    end
```

Comment:
Another interesting test will be one that test's that `temp_token` is executed `:before_create` :wink: 
That is:
```ruby
it { expect(user.token).to be nil}
user.save
it { expect(user.reload.token).not_to be nil }
```